MIPs/mip-x.md

```diff
@@ -0,0 +1,218 @@
+
+## Motivation
+
+This proposal aims to solve a key pain point for dApp developers and users - the inability to initiate swap tokens from the dApp interface.
+
+Currently, if a dApp requires users to swap from token A to token B to perform a specific operation, the user has to exit the dApp, go to MetaMask or third parties, search for the token the user needs and execute the swap, then return to the dApp. This disjointed workflow results in a poor user experience.
+
+Implementing `wallet_swapAsset` would allow the entire token swap process to occur seamlessly by navigating from the dApp to MetaMask wallet with the tokens information without requiring external integration. This provides the following benefits:
+
+- Improved usability - Users can get the right tokens for dApp operations in fewer steps and with less context switching.
+- Better dApp integration - dApps can build and control the swap experience natively with just a few lines.
+- Increased adoption - More intuitive swap workflow lowers barrier to dApp usage.
+- Interoperability - Standard method works across different wallets.
+- Increase revenue - Easier adoption means more users will use MetaMask swaps.
+
+Introducing this API directly in the wallet is preferable when compared with the alternative of using snaps, as the latter would create more friction and less adoption since users would need to install that specific snap to be able to perform a swap action. The goal is to reduce friction to swaps.
+
+Overall, this proposal fills a clear gap in dApp capabilities related to swapping tokens. Implementing it would provide significant value to both developers and users in the Ethereum ecosystem.
+
+# Usage Example
+
+```
+ethereum.request({
+  method: 'wallet_swapAsset',
+  params: [{
+    fromToken: [{
+      address: 'eip155:1:0x1234567890abcdefABCDEF1234567890ABCDEF',
+      value: '0xDE0B6B3A7640000',
+    }],
+    toToken: {
+      address: 'eip155:1:0xabcdef1234567890ABCDEF1234567890abcdef',
+    },
+    userAddress: 'eip155:1:0x0000000000000000000000000000000000000000'
+  },
+  ]
+});
+
+```
+
+## Proposal Specification
+
+The new JSON-RPC method wallet_swapAsset should be implemented with the following parameters:
+
+- `fromToken`: An object containing details about the source token. It should include:
+
+  - `address`: The CAIP-10 formatted address of the source token. This address combines the CAIP-2 blockchain identifier and the token's address on that blockchain.
+  - `value`: The amount of wei in hexadecimal format of the source token to be swapped.
+
+- `toToken`: An object containing details about the destination token. It should include:
+
+  - `address`: The CAIP-10 formatted address of the source token. This address combines the CAIP-2 blockchain identifier and the token's address on that blockchain.
```

Comment:
nit: probably don't need the second sentence here since its the same above
```suggestion
  - `address`: The CAIP-10 formatted address of the source token.
```